src/frontend/src/handler/create_sql_function.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use itertools::Itertools;
+use pgwire::pg_response::StatementType;
+use risingwave_common::catalog::FunctionId;
+use risingwave_common::types::DataType;
+use risingwave_pb::catalog::function::{Kind, ScalarFunction, TableFunction};
+use risingwave_pb::catalog::Function;
+use risingwave_sqlparser::ast::{
+    CreateFunctionBody, FunctionDefinition, ObjectName, OperateFunctionArg,
+};
+use risingwave_sqlparser::parser::Parser;
+
+use super::*;
+use crate::catalog::CatalogError;
+use crate::{bind_data_type, Binder};
+
+/// Simple type check for sql udf
+/// e.g., return type mismatch check, parameter type check, ..., etc.
+/// TODO: any existing module / function that can achieve the same functionality?
+fn _sql_udf_type_checker(ast: Vec<Statement>) -> Result<()> {
```

Comment:
we can reuse the binder later.

Reply:
I'm considering how to implement the semantic check at present.
In general the logic would be create a dummy / mock context for binder (`Binder::new_for_system`) to bind the expression, for example if the sql udf is anonymous, then create the `ParameterTypes` with the same types as pre-defined. Same logic for named sql udf.
But overall yeah, I think we could reuse the logic within binder.